frontend/src/main/java/starter/project/frontend/CarController.java

```diff
@@ -39,6 +40,7 @@ public Car getCar(@PathVariable("carId") int carId) {
 
     @PostMapping(path = "/cars", consumes = {"application/json"})
     public Map<Integer, Car> createCarList(@RequestBody List<Car> carList) {
+        List<Car> currentCarList = new ArrayList<>();
```

Comment:
This list doesn't need to be here. You now have two equal variables in the same scope. `carList` only exists in the method as does `currentCarList` and they are equivalent (until you update/change one or the other)